lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -2645,11 +2646,59 @@ def test_index_with_gated_sequential(self):
                 }
             )
         )
-
         self.assertEquals(response.status_code, 200)
         self.assertIn("Content Locked", response.content)
 
 
+@attr(shard=5)
+class TestIndexViewWithCourseDurationLimits(ModuleStoreTestCase):
+    """
+    Test the index view for a course with course duration limits enabled.
+    """
+
+    def setUp(self):
+        """
+        Set up the initial test data.
+        """
+        super(TestIndexViewWithCourseDurationLimits, self).setUp()
+
+        self.user = UserFactory()
+        self.course = CourseFactory.create(start=datetime.now() - timedelta(weeks=1))
+        with self.store.bulk_operations(self.course.id):
+            self.chapter = ItemFactory.create(parent=self.course, category="chapter")
+            self.sequential = ItemFactory.create(parent=self.chapter, category='sequential')
+
+        CourseEnrollmentFactory(user=self.user, course_id=self.course.id)
+
+    @override_waffle_flag(CONTENT_TYPE_GATING_FLAG, True)
+    def test_index_with_course_duration_limits(self):
+        """
+        Test that the courseware contains the course expiration banner
+        when course_duration_limits are enabled.
+        """
+        self.assertTrue(self.client.login(username=self.user.username, password='test'))
+        response = self.client.get(
+            reverse(
+                'courseware_section',
+                kwargs={
+                    'course_id': unicode(self.course.id),
+                    'chapter': self.chapter.url_name,
+                    'section': self.sequential.url_name,
+                }
+            )
+        )
+
+        expiration_date = (datetime.now() + timedelta(weeks=4)).strftime('%b %-d')
+        upgrade_link = verified_upgrade_deadline_link(user=self.user, course=self.course)
+        bannerText = 'Your access to this course expires on {expiration_date}. \
+            <a href="{upgrade_link}">Upgrade now</a> for unlimited access.'.format(
```

Comment:
Does it make sense to test the exact text here? It feels like that will make the test brittle if we start making copy changes, which I'm not sure we want.

Reply:
Agreed that it is brittle, but it would be easy to fix the tests along with copy changes that shouldn't happen too frequently.  
But if I just tested for the existence of the banner for example, I'm concerned that the date could be wrong or the link could be missing and so on. (For instance, the test had the wrong date that I spent a while fixing. If I just tested for a smaller substring I wouldn't have noticed that)   
Theoretically, if I had separate unit tests for how all those key things are set I could skip testing the text, but in this case it is more like one functional unit test instead of several smaller scope unit tests

Reply:
might be nice to have a method that could be used by other test classes that generated the expected banner text. I'll need this exact thing in the test I'm writting for Progress page. Then at least if there are text copy changes or changes to  the expiration logic we'll only have to change it one place rather than in each individual test.

Reply:
good idea, I'll try doing this